modules/theme/plugin.js

```diff
@@ -0,0 +1,44 @@
+import Vue from 'vue'
+import { parse, serialize } from 'cookie'
+
+const key = '<%= options.cookie.key %>'
+const cookieOptions = <%= JSON.stringify(options.cookie.options, null, 2) %>
+
+export default function (ctx, inject) {
+  const theme = Vue.observable({
+    value: (process.client ? window.__nuxt_theme : '<%= options.value %>')
+  })
+  const setTheme = (value, persist = true) => {
+    theme.value = value
+    if (process.client) {
+      // update cookie
+      if (persist) {
+        document.cookie = serialize(key, value, cookieOptions)
```

Comment:
Is it really required to use an external package (800B - 16ms) to serialize a string field? We can use a string template and also consider router.base for theme scope.